src/parquet/file_reader.cc

```diff
@@ -180,42 +243,104 @@ class SerializedFile : public ParquetFileReader::Contents {
         source_->ReadAt(file_size - footer_read_size, footer_read_size, footer_buffer);
 
     // Check if all bytes are read. Check if last 4 bytes read have the magic bits
-    if (bytes_read != footer_read_size ||
-        memcmp(footer_buffer + footer_read_size - 4, PARQUET_MAGIC, 4) != 0) {
-      throw ParquetException("Invalid parquet file. Corrupt footer.");
-    }
+    // no encryption
+    if (bytes_read == footer_read_size &&
+        memcmp(footer_buffer + footer_read_size - 4, PARQUET_MAGIC, 4) == 0) {
+      uint32_t metadata_len =
+          *reinterpret_cast<uint32_t*>(footer_buffer + footer_read_size - FOOTER_SIZE);
+      int64_t metadata_start = file_size - FOOTER_SIZE - metadata_len;
+      if (FOOTER_SIZE + metadata_len > file_size) {
+        throw ParquetException(
+            "Invalid parquet file. File is less than "
+            "file metadata size.");
+      }
 
-    uint32_t metadata_len =
-        *reinterpret_cast<uint32_t*>(footer_buffer + footer_read_size - FOOTER_SIZE);
-    int64_t metadata_start = file_size - FOOTER_SIZE - metadata_len;
-    if (FOOTER_SIZE + metadata_len > file_size) {
-      throw ParquetException(
-          "Invalid parquet file. File is less than "
-          "file metadata size.");
+      std::shared_ptr<PoolBuffer> metadata_buffer =
+          AllocateBuffer(properties_.memory_pool(), metadata_len);
```

Comment:
PoolBuffer is a private API as of Arrow 0.10.0, change this to `Buffer` (or ResizableBuffer if needed)